daemon/cmd/agenthealth.go

```diff
@@ -73,18 +77,37 @@ func (d *Daemon) startAgentHealthHTTPService(addr string) {
 		}
 		w.WriteHeader(statusCode)
 	}))
-	srv := &http.Server{
-		Addr:    addr,
-		Handler: mux,
-	}
 
-	go func() {
-		err := srv.Serve(ln)
-		if errors.Is(err, http.ErrServerClosed) {
-			log.Info("healthz status API server shutdown")
+	available := len(hosts)
+	for _, host := range hosts {
+		lc := net.ListenConfig{Control: setsockoptReuseAddrAndPort}
+		addr := net.JoinHostPort(host, fmt.Sprintf("%d", port))
+		ln, err := lc.Listen(context.Background(), "tcp", addr)
+		if errors.Is(err, unix.EADDRNOTAVAIL) {
+			log.Infof("healthz status API server not available on %s", addr)
+			available--
+			continue
 		} else if err != nil {
-			log.WithError(err).Fatal("Unable to start healthz status API server")
+			log.WithError(err).Fatalf("Unable to listen on %s for healthz status API server", addr)
```

Comment:
will log.Fatalf here will lead to exit(1) here? If yes, the loop might not continue.

Reply:
Good point, changed.

Reply:
in this case, `continue` in next line can be removed to avoid any confusion?

Reply:
Good point, done.